src/github.ts

```diff
@@ -774,23 +774,25 @@ export class GitHub {
    */
   getFileContentsWithDataAPI = wrapAsync(
     async (path: string, branch: string): Promise<GitHubFileContents> => {
-      const repoTree = await this.octokit.git.getTree({
-        owner: this.repository.owner,
-        repo: this.repository.repo,
-        tree_sha: branch,
```

Comment:
I'm not sure, but couldn't we add `recursive: 'true'` to `getTree` to retrieve all sub directories and file?

Reply:
It depends on how big the repo is - there's a limit to how much is returned (looks like [unofficially 100k entries or 7 MB of entries](https://github.community/t/github-get-tree-api-limits-and-recursivity/1300/2)). If we hit the limit, then `truncated` is returned in the response and there is no pagination options available.

So we could first attempt to fetch the entire tree, then fall back to our recursive method. We'll still need this implementation as the fallback and we could add the shortcut as an improvement.

Another alternative is to try to use the GraphQL API which I'm looking into for #1027.

Reply:
If the GraphQL API could solve both #1027 and this, it would be my preference. My only concern -- my only concern would be if it introduces oddities, e.g., I think we found search is sometimes eventually consistent, or we've at least found some API calls to sometimes be eventually consistent?

This is a fundamental part of the codebase, so perhaps we could publish #1027 to a tag for testing in some real world repos?